### PR TITLE
General maintainence for Ore Silo

### PR DESCRIPTION
**Pull request by SyncIt21**
## About The Pull Request
1. **Qol**(Hopefully not an issue for feature freeze)
   - Adds Screen tips & examines for screwdriver, crowbar & multitool acts
   - Examining ore silo will now display its stored materials
   - Location name is attached to the machine name and it greys out when pausing the machine to make it more noticable that it is paused
   - Made window wider

2. **Code Improvements**
   - Auto docs procs & vars for ore silo log entry
   - Fixed return values of all tool acts i.e. no more returning `TRUE/FALSE` but the actual item interaction flags
   - UI no longer computes `REF` for each machine which is expensive but instead uses the index value in the list
   - Moved global variables to their corresponding files
   - Ore silo no longer starts processing round start. That proc just ended itself anyway so why even bother registering

3. **Fixes**
    - If a machine was disconnected via the ore silo UI, attempting to reconnect that machine would cause a runtime at `RegisterSignal` in `multitool_act` because it was already registered in `Initialize`. It now unregisters the signal when disconnecting so that's fixed
    - Each machine element in the UI does not have a unique key because it used `key={machine.name}`. This meant after disconnecting a machine the UI buttons would get grouped together and the pause & disconnect buttons would not function properly. Now it uses `index` which is unique thus fixing that problem

## Changelog
:cl:
qol: added more screen tips & examines for ore silo, made UI wider, attach location name to each machine & grey out paused machines to make it more noticeable.
code: auto docs proc & vars for silo log entry. Fixed return values of all tool acts
fix: ore silo UI now functions correctly after removing an entry from the UI
fix: no runtimes when connecting a machine to silo that was previously disconnected via the ore silo UI
/:cl:

### CODE REVIEW

File: code/modules/mining/machine_silo.dm [hunk 1]
```diff
@@ -92,39 +124,35 @@ GLOBAL_LIST_EMPTY(silo_access_logs)
 	return materials.ui_static_data()
 
 /obj/machinery/ore_silo/ui_data(mob/user)
-	var/list/data = list(
-		"materials" =  materials.ui_data()
-	)
+	. = list()
```
Comment (JohnFulpWillard) [inline code/modules/mining/machine_silo.dm:127]: Revert back to using ``var/list/data`` please, that's what our standards say to use

File: code/modules/mining/machine_silo.dm [hunk 2]
```diff
@@ -133,31 +163,53 @@ GLOBAL_LIST_EMPTY(silo_access_logs)
...
-				return TRUE
+			if(!istype(ejecting))
+				return
+
+			var/amount = params["amount"]
+			if(isnull(amount))
+				return
+
+			amount = text2num(amount)
+			if(isnull(amount))
```
Comment (intercepti0n) [inline code/modules/mining/machine_silo.dm:207]: Why do you use `isnull` instead of `isnum`?

Comment (SyncIt21) [reply code/modules/mining/machine_silo.dm:207]: because test2num will already do that check for us internally & if it fails it returns null

**Comment by OverwatchVoice** [comment]
These changes have been very nice, but the lack of pages for the ore silo logs means that if the round has been going for some time, the list gets so long that your game grinds to a complete halt when you open them!

**Comment by intercepti0n** [comment]
@OverwatchVoice , thanks, as a former author of the UI I'll PR a fix by tomorrow.

**Pull request merged**